app/scripts/lib/createDupeReqFilterMiddleware.ts

```diff
@@ -0,0 +1,86 @@
+import { JsonRpcMiddleware } from 'json-rpc-engine';
+import log from 'loglevel';
+
+const SECOND = 1000;
+const MINUTE = SECOND * 60;
+export const THREE_MINUTES = MINUTE * 3;
+
+/**
+ * Creates a set abstraction whose values expire after five minutes.
+ *
+ * @returns The expiry set.
+ */
+const makeExpirySet = () => {
+  const seenValues: (number | string)[] = [];
+  const timestamps: number[] = [];
+
+  setTimeout(() => {
```

Comment:
shouldn't this be setInterval?

Reply:
Do we want this only flush once per connection?

Reply:
lol, jesus